File: LibVisWeb/LibVisWeb/Models/News.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using LibVisLib;

namespace LibVisWeb.Models
{
    
    public class ListItem
    {

        public string Id { get; set; }
        public string Text { get; set; }

        public ListItem()
        {

            Id = "";
            Text = "";
 
        }
        
        public ListItem(string i, string t)
        {

            Id = i;
            Text = t;

        }

    }

    public class SiteWarningModel
    {

        public int Type { get; set; }
        public string Title { get; set; }
        public string Text { get; set; }

        public SiteWarningModel()
        {

            Type = 0;
            Title = "";
            Text = "";

        }

        public SiteWarningModel(LibVisLib.SiteWarning s)
        {

            Type = (int)s.type;
            Title = s.title;
            Text = s.text;

        }

    }

    public class HomePageModel
    {

        public List<VideoModel> Videos { get; set; }
        public List<NewsStatisticModel> Statistics { get; set; }
        public List<ArticleRowModel> Articles { get; set; }
        public List<SiteWarningModel> Warnings { get; set; }

        public int NumTargets { get; set; }
        public int NumApproval { get; set; }
        public int NumRevision { get; set; }
        public int NumNarration { get; set; }
        public int NumProduction { get; set; }

        public int Seq { get; set; }

        public HomePageModel()
        {

            Videos = new List<VideoModel>();
            Statistics = new List<NewsStatisticModel>();
            Articles = new List<ArticleRowModel>();
            Warnings = new List<SiteWarningModel>();
            
            NumTargets = 0;
            NumApproval = 0;
            NumRevision = 0;
            NumNarration = 0;
            NumProduction = 0;

            Seq = 0;

        }

    }

    public class MainPageModel
    {

        public List<TargetModel> Targets { get; set; }
        public List<ArticleModel> Articles { get; set; }
        public List<VideoModel> Videos { get; set; }

        public int NumArticles { get; set; }
        public int NumShortNote { get; set; }
        public int NumSatoshis { get; set; }

        public MainPageModel()
        {

            Targets = new List<TargetModel>();
            Articles = new List<ArticleModel>();
            Videos = new List<VideoModel>();
            NumArticles = 0;
            NumShortNote = 0;
            NumSatoshis = 0;

        }

    }

    public class UserPageModel
    {

        public AccountModel User { get; set; }
        public List<TargetModel> Targets { get; set; }
        public List<ArticleModel> Articles { get; set; }
        public List<VideoModel> Videos { get; set; }
        
        public UserPageModel()
        {

            User = new AccountModel();
            Targets = new List<TargetModel>();
            Articles = new List<ArticleModel>();
            Videos = new List<VideoModel>();

        }

    }
    
    public class CategInfo
    {

        public NewsCategory MainCategory { get; set; }
        public List<NewsCategory> Categories { get; set; }

        public CategInfo()
        {

            MainCategory = new NewsCategory();
            Categories = new List<NewsCategory>();

        }

    }

    public class AuthorInfo 
    {

        public string SuggestedLabel { get; set; }
        public string AuthoredLabel { get; set; }
        public string RevisedLabel { get; set; }
        public string NarratedLabel { get; set; }
        public string ProducedLabel { get; set; }
                
        public UserIdModel Suggested { get; set; }
        public UserIdModel Authored { get; set; }
        public UserIdModel Revised { get; set; }
        public UserIdModel Narrated { get; set; }
        public UserIdModel Produced { get; set; }

        public string DateLabel { get; set; }
        public string Date { get; set; }

        public string StatusText { get; set; }
        
        public AuthorInfo()
        {

            SuggestedLabel = "";
            AuthoredLabel = "";
            RevisedLabel = "";
            NarratedLabel = "";
            ProducedLabel = "";

            Suggested = new UserIdModel();
            Authored = new UserIdModel();            
            Revised = new UserIdModel();
            Narrated = new UserIdModel();
            Produced = new UserIdModel();

            DateLabel = "";
            Date = "";

            StatusText = "";

        }

    }

    public class NewsCategory
    {

        public string Label { get; set; }
        public string Category { get; set; }

        public NewsCategory()
        {

            Label = "";
            Category = "";

        }

        public NewsCategory(RacLib.RacMsg msgs, LibVisLib.Category c)
        {

            Label = c.label;
            Category = msgs.Get(c.nameMsg);

        }

    }

    public class NewsAward
    {

        public string Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }

        public NewsAward()
        {

            Id = "";
            Name = "";
            Description = "";

        }

        public NewsAward(LibVisLib.Award a)
        {

            Id = a.id;
            Name = a.name;
            Description = a.description;

        }

    }

    public class NewsStatisticModel
    {

        public string Realm { get; set; }
        public string Parameter { get; set; }
        public string ImageLink { get; set; }
        public int Value { get; set; }
        public string Units { get; set; }
        public string Link { get; set; }

        public NewsStatisticModel(RacLib.RacMsg msgs, Statistics ns)
        {

            switch (ns.realm)
            {

                case Statistics.Realm.Youtube:
                    Realm = "Youtube";
                    break;

                case Statistics.Realm.Bitchute:
                    Realm = "Bitchute";
                    break;

                case Statistics.Realm.Facebook:
                    Realm = "Facebook";
                    break;

                case Statistics.Realm.Minds:
                    Realm = "Minds";
                    break;

                case Statistics.Realm.Twitter:
                    Realm = "Twitter";
                    break;

                case Statistics.Realm.Gabai:
                    Realm = "Gab";
                    break;

                case Statistics.Realm.Site:
                    Realm = msgs.Get(RacLib.RacMsg.Id.LibertarianViewSiteName);
                    break;

            }

            switch (ns.parameter)
            {

                case Statistics.Parameter.Subscribers:
                    Parameter = msgs.Get(RacLib.RacMsg.Id.Subscribers);
                    break;

                case Statistics.Parameter.Views:
                    Parameter = msgs.Get(RacLib.RacMsg.Id.Views);
                    break;

            }

            ImageLink = ns.icon;
            Value = ns.value;
            Units = ns.units;
            Link = ns.link;

        }

    }

    public class UserIdModel
    {

        public string Name { get; set; }
        public string Id { get; set; }

        public UserIdModel()
        {

            Name = "";
            Id = "";

        }

        public UserIdModel(string i, string n)
        {

            Name = n;
            Id = i;

        }

    }

    public class RegisterGrant
    {

        public string ActionId { get; set; }
        public string AwardId { get; set; }
        public int Add { get; set; }
        public int Lang { get; set; }

        public RegisterGrant()
        {

            ActionId = "";
            AwardId = "";
            Add = 0;
            Lang = 0;

        }

    }
    
    /// <summary>
    /// Registro da newsletter
    /// </summary>
    public class NewsLetterRegister
    {

        public int Type { get; set; }
        public string Data { get; set; }
        public int Lang { get; set; }

        public NewsLetterRegister()
        {

            Type = 0;
            Data = "";
            Lang = 0;

        }

    }

    public class ChangeLanguage
    {

        public string Id { get; set; }
        public int Lang { get; set; }
        public int NewLang { get; set; }

        public ChangeLanguage()
        {

            Id = "";
            Lang = 0;
            NewLang = 0;

        }

    }
    
}

File: LibVisWeb/LibVisWeb/ClientApp/models/News.ts
import { AccountModel } from './Account';
import { VideoModel } from './Video';
import { ArticleModel, ArticleRowModel } from './Article';
import { TargetModel } from './Target';

//
//  The classes here should match webapi models
//

export class HomePageModel {

    public Videos: VideoModel[] = [];
    public Statistics: NewsStatisticsModel[] = [];
    public Articles: ArticleRowModel[] = [];
    public Warnings: SiteWarningModel[] = [];
    
    public NumTargets: number = 0;
    public NumApproval: number = 0;
    public NumRevision: number = 0;
    public NumNarration: number = 0;
    public NumProduction: number = 0;

    public Seq: number = 0;

}

export class SiteWarningModel {

    public Type: number = 0;
    public Title: string = "";
    public Text: string = "";

}

export class MainPageModel {

    public Targets: TargetModel[] = [];
    public Articles: ArticleModel[] = [];
    public Videos: VideoModel[] = [];

    public NumArticles: number = 0;
    public NumShortNote: number = 0;
    public NumSatoshis: number = 0;

}

export class NewsStatisticsModel {

    public Realm: string = "";
    public Parameter: string = "";
    public ImageLink: string = "";
    public Value: number = 0;
    public Units: string = "";
    public Link: string = "";

}

export class UserIdModel {

    public Id: string = "";
    public Name: string = "";
    
}

export class NewsCategory {

    public Label: string = "";
    public Category: string = "";

}

export class NewsAward {

    public Id: string = "";
    public Name: string = "";
    public Description: string = "";

}

export class CategInfo {

    public MainCategory: NewsCategory = { Label: "", Category: "" };
    public Categories: NewsCategory[] = [];

}

export class AuthorInfo {
    
    public SuggestedLabel: string = "";
    public AuthoredLabel: string = "";
    public RevisedLabel: string = "";
    public NarratedLabel: string = "";
    public ProducedLabel: string = "";

    public Suggested: UserIdModel = { Id: "", Name: "" };
    public Authored: UserIdModel = { Id: "", Name: "" };
    public Revised: UserIdModel = { Id: "", Name: "" };
    public Narrated: UserIdModel = { Id: "", Name: "" };
    public Produced: UserIdModel = { Id: "", Name: "" };

    public DateLabel: string = "";
    public Date: string = "";

    public StatusText: string = "";

}

export class ListItem {

    public Id: string = "";
    public Text: string = "";

}

export class UserPageModel {

    public User: AccountModel = { Id: "", Email: "", Name: "", Bitcoin: "", Profile: 0, ProfileName: "", Lang: 2, Medals: [], Payments: [], Videos: [], Value: 0, Xp: 0, NextXp: 0, Ap: 0, NextAp: 0, Rp: 0, NextRp: 0, Op: 0, NextOp: 0, Qp: 0, NextQp: 0, Tp: 0, NextTp: 0, Up: 0, NextUp: 0, Sp: 0, NextSp: 0, Ep: 0, NextEp: 0, NewsLetter: false, GeneralQualification: "", TotalQualification: "", EditorQualification: "", ReporterQualification: "", RoleTranslatorEn: 0, RoleTranslatorPt: 0, RoleTranslatorEs: 0, RoleRevisorEn: 0, RoleRevisorPt: 0, RoleRevisorEs: 0, RoleNarratorEn: 0, RoleNarratorPt: 0, RoleNarratorEs: 0, RoleProducerEn: 0, RoleProducerPt: 0, RoleProducerEs: 0, Translator: 0, Revisor: 0, Narrator: 0, Producer: 0, Sponsor: 0, Staff: 0, Admin: 0, Blocked: 0, NotConfirmed: 0 };

    public Targets: TargetModel[] = [];
    public Articles: ArticleModel[] = [];
    public Videos: VideoModel[] = [];

}

export class RegisterGrant {

    public ActionId: string = "";
    public AwardId: string = "";
    public Add: number = 0;
    public Lang: number = 0;

}

export class NewsLetterRegister {

    public Type: number = 0;
    public Data: string = "";
    public Lang?: number = 0;

}

export class ChangeLanguage {

    public Id?: string = "";
    public Lang?: number = 0;
    public NewLang?: number = 0;

}
